### PR TITLE
TMDM-10860 Fixed "New Value" can't get real type in update report.

### PR DESCRIPTION
**Pull request by yjli-talend**

### CODE REVIEW

File: org.talend.mdm.core/src/com/amalto/core/save/context/UpdateActionCreator.java [hunk 1]
```diff
@@ -501,6 +506,18 @@ public void execute(FieldMetadata field) {
                 Accessor rightAccessor = newDocument.createAccessor(getRightPath());
                 if (rightAccessor.exist()) {
                     String newType = rightAccessor.getActualType();
+                    if (field instanceof ReferenceFieldMetadata) {
+                        ComplexTypeMetadata typeMetadata = repository.getComplexType(newType);
+                        if (typeMetadata != null && typeMetadata.getSubTypes().size() > 0) {
+                            String value = rightAccessor.get();
+                            if (value != null && !value.isEmpty()) {
```
Comment (pbabin) [inline org.talend.mdm.core/src/com/amalto/core/save/context/UpdateActionCreator.java]: Could use StringUtils utility method instead

File: org.talend.mdm.core/src/com/amalto/core/save/context/UpdateActionCreator.java [hunk 2]
```diff
@@ -501,6 +506,18 @@ public void execute(FieldMetadata field) {
...
+                                String actualType = getActualType(typeMetadata,value);
+                                if (!actualType.isEmpty()) {
```
Comment (pbabin) [inline org.talend.mdm.core/src/com/amalto/core/save/context/UpdateActionCreator.java]: Same here using StringUtils

File: org.talend.mdm.core/src/com/amalto/core/save/context/UpdateActionCreator.java [hunk 3]
```diff
@@ -539,5 +556,39 @@ public void execute(FieldMetadata field) {
                 }
             }
         }
+        
+        private String getActualType(ComplexTypeMetadata type,String ids) {
+            String realTypeName = "";
```
Comment (pbabin) [inline org.talend.mdm.core/src/com/amalto/core/save/context/UpdateActionCreator.java]: Useless initialization

**Comment by pbabin** [comment]
Missing detailed description about what are the changes to correct the issue

**Comment by lpw82gf** [comment]
LGTM

**Pull request merged**